collector/src/main/java/com/bitmovin/analytics/features/errordetails/ErrorDetailTracking.kt

```diff
@@ -0,0 +1,48 @@
+package com.bitmovin.analytics.features.errordetails
+
+import android.content.Context
+import com.bitmovin.analytics.data.AdEventData
+import com.bitmovin.analytics.data.EventData
+import com.bitmovin.analytics.features.Feature
+import com.bitmovin.analytics.features.segmenttracking.SegmentTracking
+import com.bitmovin.analytics.utils.topOfStacktrace
+
+class ErrorDetailTracking(val context: Context, private val segmentTracking: SegmentTracking?, private vararg val eventSources: OnErrorDetailEventSource) : Feature<ErrorDetailTrackingConfig>(), OnErrorDetailEventListener {
+    override val name = "errorDetails"
+    override val configClass = ErrorDetailTrackingConfig::class.java
+
+    private val backend = ErrorDetailBackend(context)
+
+    init {
+        eventSources.forEach { it.addEventListener(this) }
+    }
+
+    override fun configure(authenticated: Boolean, config: ErrorDetailTrackingConfig?) {
+        // We need to make sure that the features we depend on have already be configured.
+        // In this case, `SegmentTracking` already needs to have the `maxSegments` set to the correct value.
+        if (segmentTracking != null) {
+            backend.queue.forEach {
+                if (it.segments == null)
+                    return@forEach
+                while (it.segments.size > segmentTracking.maxSegments) {
+                    it.segments.removeAt(it.segments.size - 1)
```

Comment:
is this working?
removing elements from list while in `while`?
Could this imply any troubles?

Reply:
🤔 yes this might be an issue removing the segment while going through it.
also.. clamping to the max segment size should not be done in the configure method but rather in a separate method

Reply:
isn't there a `limitQueue` method?